src/Compiler/Checking/infos.fs

```diff
@@ -1832,6 +1832,38 @@ type PropInfo =
         | ProvidedProp(_, pi, m) -> pi.PUntaint((fun pi -> pi.CanWrite), m)
 #endif
 
+    member x.GetterAccessibility =
+        match x with
+        | ILProp ilpinfo when ilpinfo.HasGetter -> Some taccessPublic
+        | ILProp _ -> None
+
+        | FSProp(_, _, Some getter, _) -> Some getter.Accessibility
+        | FSProp _ -> None
+
+#if !NO_TYPEPROVIDERS
```

Comment:
Can you please remind me why type providers have a special case here?

Reply:
Because it's provided property, which exists only for type providers.

Reply:
> Can you please remind me why type providers have a special case here?您能提醒我为什么类型提供商在这里有特殊情况吗？

```fsharp
    /// Indicates if this property has an associated getter method.
    member x.HasGetter =
        match x with
        | ILProp ilpinfo-> ilpinfo.HasGetter
        | FSProp(_, _, x, _) -> Option.isSome x
#if !NO_TYPEPROVIDERS
        | ProvidedProp(_, pi, m) -> pi.PUntaint((fun pi -> pi.CanRead), m)
#endif
```